apps/web/vibes/soul/sections/address-list-section/index.tsx

```diff
@@ -141,132 +127,146 @@ export function AddressListSection<A extends Address>({
       <div>
         {showNewAddressForm && (
           <div className="border-b border-contrast-200 pb-6 pt-5">
-            <AddressForm
-              action={formAction}
-              address={{
-                id: 'new',
-                firstName: '',
-                lastName: '',
-                street1: '',
-                street2: '',
-                city: '',
-                state: '',
-                country: '',
-              }}
-              addressLevel1Label={addressLevel1Label}
-              addressLevel2Label={addressLevel2Label}
-              addressLine1Label={addressLine1Label}
-              addressLine2Label={addressLine2Label}
-              cancelLabel={cancelLabel}
-              companyLabel={companyLabel}
-              countryLabel={countryLabel}
-              firstNameLabel={firstNameLabel}
-              intent="create"
-              lastNameLabel={lastNameLabel}
-              onCancel={() => setShowNewAddressForm(false)}
-              onSubmit={(formData) => {
-                setShowNewAddressForm(false);
-
-                startTransition(() => {
-                  formAction(formData);
-                  setOptimisticState(formData);
-                });
-              }}
-              phoneLabel={phoneLabel}
-              postalCodeLabel={postalCodeLabel}
-              submitLabel={createLabel}
-            />
-          </div>
-        )}
-        {optimisticState.addresses.map((address) => (
-          <div className="border-b border-contrast-200 pb-6 pt-5" key={address.id}>
-            {activeAddressIds.includes(address.id) ? (
-              <AddressForm
-                action={formAction}
-                address={address}
-                addressLevel1Label={addressLevel1Label}
-                addressLevel2Label={addressLevel2Label}
-                addressLine1Label={addressLine1Label}
-                addressLine2Label={addressLine2Label}
-                cancelLabel={cancelLabel}
-                companyLabel={companyLabel}
-                countryLabel={countryLabel}
-                firstNameLabel={firstNameLabel}
-                intent="update"
-                lastNameLabel={lastNameLabel}
-                onCancel={() =>
-                  setActiveAddressIds((prev) => prev.filter((id) => id !== address.id))
-                }
-                onSubmit={(formData) => {
-                  setActiveAddressIds((prev) => prev.filter((id) => id !== address.id));
+            <div className="w-[480px] space-y-4">
+              <DynamicForm
+                action={(_prevState, formData) => {
```

Comment:
For progressive enhancement, we should use the form action directly or from useActionState and use onSubmit for optimistic logic